src/SIL.Harmony.Core/ObjectSnapshot.cs

```diff
@@ -0,0 +1,53 @@
+using System.Diagnostics.CodeAnalysis;
+using System.Text.Json.Serialization;
+
+namespace SIL.Harmony.Core;
+
+public class ObjectSnapshot
```

Comment:
it looks like actually moving this would be more painful than I expected. There's also the fact that since it's stored in the Db changing the Commit type might cause problems there. Instead lets just extract an interface. It should have all the properties but the Commit type will be CommitBase on the interface. We then need to make an explicit interface implementation on ObjectSnapshot. That looks like this:
`CommitBase IObjectSnapshot.Commit => Commit`, this basically makes this Commit property only get used if the object is referred to as `IObjectSnapshot` if it's referred to as `ObjectSnapshot` then it will use the public property.

Reply:
Done in commit af8d9de